silence/silence.go

```diff
@@ -564,6 +564,13 @@ func (s *Silences) getSilence(id string) (*pb.Silence, bool) {
 	return msil.Silence, true
 }
 
+func (s *Silences) toMeshSilence(sil *pb.Silence) *pb.MeshSilence {
```

Comment:
I've moved this to a function for two reasons:

1. It allows us to check the full error in unit tests. For example:

    ```go
    require.EqualError(t, s.Set(sil3), fmt.Sprintf("silence exceeded maximum size: %d bytes (limit: 4096 bytes)", s.toMeshSilence(sil3).Size()))
    ```

2. We will use it in the next PR where we fix the same bug but for maximum size limits and validation errors.